mobile/src/main/java/org/openhab/habdroid/background/ItemsControlsProviderService.kt

```diff
@@ -273,6 +237,73 @@ class ItemsControlsProviderService : ControlsProviderService() {
+            val typesWithState = listOf(
+                ControlTemplate.TYPE_ERROR,
+                ControlTemplate.TYPE_NO_TEMPLATE,
+                ControlTemplate.TYPE_STATELESS
+            )
+
+            return when {
+                controlTemplate.templateType !in typesWithState -> null
+                !item.options.isNullOrEmpty() -> {
+                    item.options
+                        .firstOrNull { labeledValue -> labeledValue.value == item.state?.asString }
+                        ?.label
+                }
+                item.isOfTypeOrGroupType(Item.Type.Number) -> item.state?.asNumber?.toString()
+                else -> item.state?.asString
+            }
+        }
+
+        private fun getIntent(item: Item) = when {
+            !item.readOnly && item.options != null -> {
+                val intent = Intent(context, SelectionItemActivity::class.java).apply {
+                    putExtra(SelectionItemActivity.EXTRA_ITEM, item)
+                }
+                Pair(intent, item.hashCode())
+            }
+            !item.readOnly && item.isOfTypeOrGroupType(Item.Type.Color) -> {
+                val intent = Intent(context, ColorItemActivity::class.java).apply {
+                    putExtra(ColorItemActivity.EXTRA_ITEM, item)
+                }
+                Pair(intent, item.hashCode())
+            }
+            else -> {
+                val intent = Intent(context, MainActivity::class.java).apply {
+                    putExtra(MainActivity.EXTRA_SERVER_ID, primaryServerId)
+                }
+                Pair(intent, primaryServerId)
+            }
+        }
+
+        private fun getControlTemplate(item: Item): ControlTemplate {
+            return when {
+                item.readOnly -> {
+                    ControlTemplate.getNoTemplateObject()
+                }
+                item.options != null -> {
+                    // Open app when clicking on tile
+                    ControlTemplate.getNoTemplateObject()
+                }
+                item.type == Item.Type.Switch -> ToggleTemplate(
+                    item.name,
+                    ControlButton(item.state?.asBoolean ?: false, context.getString(R.string.nfc_action_toggle))
+                )
+                item.type == Item.Type.Dimmer || item.type == Item.Type.Color -> ToggleRangeTemplate(
+                    "${item.name}_toggle",
+                    ControlButton(item.state?.asBoolean ?: false, context.getString(R.string.nfc_action_toggle)),
+                    createRangeTemplate(item, "%.0f%%")
+                )
+                item.type == Item.Type.Rollershutter -> createRangeTemplate(item, "%.0f%%")
+                item.type == Item.Type.Number -> createRangeTemplate(
+                    item,
+                    item.state?.asNumber?.unit?.let { "%.0f $it" } ?: "%.0f"
+                )
+                else -> ControlTemplate.getNoTemplateObject()
```

Comment:
For some item types, like Location, no useful text-only representation is possible. For these items no tile should be created.